pkg/controller/nodeconfig/controller.go

```diff
@@ -57,6 +57,32 @@ func (c *Controller) OnNodeConfigChange(key string, nodecfg *nodeconfigv1.NodeCo
 		return nil, nil
 	}
 
+	// V2 Data Engine related handling.  This is intentionally not bothering
+	// to check whether the engine is already enabled or not, it runs on any
+	// change to the node config, even if that change wasn't related to the
+	// longhorn settings.  This is mostly harmless, because if the engine is
+	// already in the state about to be applied, re-applying that state is
+	// effectively a no-op, and I'd rather keep the code simple than add
+	// annotations for whether or not we already enabled the engine.
+	// The one wrinkle is that when allocating (or deallocating) hugepages,
+	// the kubelet needs to be restarted to pick up the change and reflect
+	// that in node.status.capacity.hugepages-2Mi, so that Longhorn can
+	// query that value when lhs/v2-data-engine is set to true.  This restart
+	// logic is handled inside EnableV2DataEngine() and DisableV2DataEngine().
```

Comment:
Do we need to handle the disabling path? If a node has v2 engine volumes, do we allow the user to disable the feature? We can implement a webhook to prohibit users from doing this in some situations. Note for an experiment feature, this looks great and we can always improve it later.

Reply:
Yeah, here in this context we do need to be able to disable the v2 engine on individual nodes, for example if you have some nodes with good fast disks and lots of RAM, and some lesser powered nodes that you don't want to use the V2 engine on (see https://longhorn.io/docs/1.7.0/v2-data-engine/features/selective-v2-data-engine-activation/).

You're right though, disabling the v2 engine globally would prevent your v2 volumes from running, thanks for pointing that out! I agree it's worth adding a check for that, but that needs to happen in the main harvester code and/or the GUI (I'll look at that separately).